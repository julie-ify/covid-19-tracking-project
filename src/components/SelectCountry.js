import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faArrowCircleRight,
  faGlobeEurope,
} from '@fortawesome/free-solid-svg-icons';

const SelectCountry = (props) => {
  const { country, cardNumber } = props;
  return (
    <div
      className={
        cardNumber === 2 || cardNumber === 3 ? 'coloredCard' : 'plainCard'
      }
    >
      <div className="countryMap">
        <FontAwesomeIcon icon={faGlobeEurope} className="map" />
        <div className="selectCountry">
          <Link to={`${country.id}`} className="regionLink">
            <FontAwesomeIcon icon={faArrowCircleRight} />
          </Link>
          <div className="countryStat">
            <div className="bigHeading">{country.name}</div>
            <div>
              {country.today_confirmed}
              {' '}
              cases
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

SelectCountry.propTypes = {
  country: PropTypes.shape({
    id: PropTypes.string,
    name: PropTypes.string,
    today_confirmed: PropTypes.number,
  }),
  cardNumber: PropTypes.number,
};

SelectCountry.defaultProps = {
  country: PropTypes.shape({
    id: '',
  }),
  cardNumber: 0,
};

export default SelectCountry;
