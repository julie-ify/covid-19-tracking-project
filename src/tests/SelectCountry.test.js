import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/ConfigureStore';
import SelectCountry from '../components/SelectCountry';

it('renders correctly', () => {
  const select = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <SelectCountry />
        </BrowserRouter>
      </Provider>,
    )
    .toJSON();
  expect(select).toMatchSnapshot();
});