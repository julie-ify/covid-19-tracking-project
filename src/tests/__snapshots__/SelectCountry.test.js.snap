// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="plainCard"
>
  <div
    className="selectCountry"
  >
    <a
      className="regionLink"
      href="/undefined"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-arrow-circle-right fa-w-16 "
        data-icon="arrow-circle-right"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8c137 0 248 111 248 248S393 504 256 504 8 393 8 256 119 8 256 8zm-28.9 143.6l75.5 72.4H120c-13.3 0-24 10.7-24 24v16c0 13.3 10.7 24 24 24h182.6l-75.5 72.4c-9.7 9.3-9.9 24.8-.4 34.3l11 10.9c9.4 9.4 24.6 9.4 33.9 0L404.3 273c9.4-9.4 9.4-24.6 0-33.9L271.6 106.3c-9.4-9.4-24.6-9.4-33.9 0l-11 10.9c-9.5 9.6-9.3 25.1.4 34.4z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </a>
    <div
      className="countryStat"
    >
      <div
        className="bigHeading"
      >
        bound checkType
      </div>
      <div>
        cases
      </div>
    </div>
  </div>
</div>
`;
